### ISSUE TITLE
drm is not being removed on kindle files

### ISSUE THREAD

**Issue by HorrorNerd** [apprenticeharper/DeDRM_tools#231]
I have kindle for pc version 1.17 and the dedrm plugin will not remove the drm from this and 4 other books i bought from amazon

Convert book 1 of 1 (Batman Vol. 1: The Court of Owls (The New 52) (Batman Graphic Novel))
Resolved conversion options
calibre version: 2.75.1
{'asciiize': False,
 'author_sort': None,
 'authors': None,
 'base_font_size': 0.0,
 'book_producer': None,
 'change_justification': u'original',
 'chapter': u"//*[((name()='h1' or name()='h2') and re:test(., '\\s*((chapter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', 'i')) or @class = 'chapter']",
 'chapter_mark': u'pagebreak',
 'comments': None,
 'cover': u'C:\\Users\\Herrings\\AppData\\Local\\Temp\\calibre_o8k5w7\\nu4ai3.jpeg',
 'debug_pipeline': None,
 'dehyphenate': True,
 'delete_blank_paragraphs': True,
 'disable_font_rescaling': False,
 'dont_compress': False,
 'duplicate_links_in_toc': False,
 'embed_all_fonts': False,
 'embed_font_family': None,
 'enable_heuristics': False,
 'expand_css': False,
 'extra_css': None,
 'extract_to': None,
 'filter_css': u'',
 'fix_indents': True,
 'font_size_mapping': None,
 'format_scene_breaks': True,
 'html_unwrap_factor': 0.4,
 'input_encoding': None,
 'input_profile': <calibre.customize.profiles.InputProfile object at 0x00000000028C12E8>,
 'insert_blank_line': False,
 'insert_blank_line_size': 0.5,
 'insert_metadata': False,
 'isbn': None,
 'italicize_common_cases': True,
 'keep_ligatures': False,
 'language': None,
 'level1_toc': None,
 'level2_toc': None,
 'level3_toc': None,
 'line_height': 0.0,
 'linearize_tables': False,
 'margin_bottom': 5.0,
 'margin_left': 5.0,
 'margin_right': 5.0,
 'margin_top': 5.0,
 'markup_chapter_headings': True,
 'max_toc_links': 50,
 'minimum_line_height': 120.0,
 'mobi_toc_at_start': False,
 'no_chapters_in_toc': False,
 'no_inline_navbars': False,
 'no_inline_toc': False,
 'output_profile': <calibre.customize.profiles.GenericEink object at 0x00000000028C1668>,
 'page_breaks_before': u"//*[name()='h1' or name()='h2']",
 'prefer_author_sort': False,
 'prefer_metadata_cover': False,
 'pretty_print': False,
 'pubdate': None,
 'publisher': None,
 'rating': None,
 'read_metadata_from_opf': u'C:\\Users\\Herrings\\AppData\\Local\\Temp\\calibre_o8k5w7\\o7nlk2.opf',
 'remove_fake_margins': True,
 'remove_first_image': False,
 'remove_paragraph_spacing': False,
 'remove_paragraph_spacing_indent_size': 1.5,
 'renumber_headings': True,
 'replace_scene_breaks': u'',
 'search_replace': '[]',
 'series': None,
 'series_index': None,
 'share_not_sync': False,
 'smarten_punctuation': False,
 'sr1_replace': None,
 'sr1_search': None,
 'sr2_replace': None,
 'sr2_search': None,
 'sr3_replace': None,
 'sr3_search': None,
 'start_reading_at': None,
 'subset_embedded_fonts': False,
 'tags': None,
 'timestamp': None,
 'title': None,
 'title_sort': None,
 'toc_filter': None,
 'toc_threshold': 6,
 'toc_title': None,
 'transform_css_rules': '[]',
 'unsmarten_punctuation': False,
 'unwrap_lines': True,
 'use_auto_toc': False,
 'verbose': 2}
InputFormatPlugin: MOBI Input running
on C:\Users\Herrings\AppData\Local\Temp\calibre_o8k5w7\njkdya.azw
Found KF8 MOBI of type 'standalone'
Python function terminated unexpectedly
  Batman Vol. 1: The Court of Owls (The New 52) (Batman Graphic Novel) (Error Code: 1)
Traceback (most recent call last):
  File "site.py", line 132, in main
  File "site.py", line 109, in run_entry_point
  File "site-packages\calibre\utils\ipc\worker.py", line 195, in main
  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 33, in gui_convert_override
  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 26, in gui_convert
  File "site-packages\calibre\ebooks\conversion\plumber.py", line 1064, in run
  File "site-packages\calibre\customize\conversion.py", line 245, in __call__
  File "site-packages\calibre\ebooks\conversion\plugins\mobi_input.py", line 44, in convert
  File "site-packages\calibre\ebooks\mobi\reader\mobi8.py", line 86, in __call__
  File "site-packages\calibre\ebooks\mobi\reader\mobi6.py", line 146, in check_for_drm
calibre.ebooks.DRMError: Batman Vol. 1: The Court of Owls (The New 52) (Batman Graphic Novel)

**Comment by ElleKayEm** [comment]
We need to see an import log to see what's happening with DRM removal.  See the FAQs:  https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Comment by HorrorNerd** [comment]
calibre Debug log
calibre 2.80 [64bit]  embedded-python: True is64bit: True
Windows-8-6.2.9200 Windows ('64bit', 'WindowsPE')
('Windows', '8', '6.2.9200')
Python 2.7.9
Windows: ('8', '6.2.9200', '', 'Multiprocessor Free')
Successfully initialized third party plugins: DeDRM (6, 5, 3) && Kindle and Mobipocket DeDRM (0, 4, 5)
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 100.861627907 x 100.55257732
Starting up...
Started up in 27.14 seconds with 59 books
DeDRM v6.5.3: Trying to decrypt B00C2IHYPW_EBOK.azw
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Preacher Book One
Found 0 keys to try after 4.0 seconds
Crypto Type is: 2
DeDRM v6.5.3: Failed to decrypt with error: No key found in 0 keys tried.
DeDRM v6.5.3: Looking for new default Kindle Key after 4.0 seconds
searching for kinfoFiles in C:\Users\Herrings\AppData\Local
Found K4PC 1.9+ kinf2011 file: C:\Users\Herrings\AppData\Local\Amazon\Kindle\storage\.kinf2011
Decrypted key file using IDString '1289721137' and UserName '48657272696e6773'
DeDRM v6.5.3: Found 1 new key
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Preacher Book One
Found 4 keys to try after 8.4 seconds
Crypto Type is: 2
File is encoded with PID SqVsJiwwQA.
Decrypting. Please wait . . . . . . . done
Decryption succeeded after 11.4 seconds
DeDRM v6.5.3: Saving 1 new key
DeDRM v6.5.3: Finished after 13.0 seconds
Using Library AlfCrypto DLL/DYLIB/SO
Running file type plugin Kindle and Mobipocket DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.k4mobidedrm.__init__", line 68, in run
  File "calibre_plugins.k4mobidedrm.__init__", line 239, in config
  File "site-packages\calibre\customize\zipplugin.py", line 183, in load_module
  File "calibre_plugins.k4mobidedrm.config", line 1, in <module>
ImportError: No module named PyQt4.Qt
Added 0 books in 26.6 seconds
DeDRM v6.5.3: Trying to decrypt B00D6LPETM_EBOK.azw
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Green Arrow Year One
Found 4 keys to try after 2.3 seconds
Crypto Type is: 2
File is encoded with PID DFwNce2jAB.
Decrypting. Please wait . . . . done
Decryption succeeded after 3.9 seconds
DeDRM v6.5.3: Finished after 4.5 seconds
Using Library AlfCrypto DLL/DYLIB/SO
Running file type plugin Kindle and Mobipocket DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.k4mobidedrm.__init__", line 68, in run
  File "calibre_plugins.k4mobidedrm.__init__", line 239, in config
  File "site-packages\calibre\customize\zipplugin.py", line 183, in load_module
  File "calibre_plugins.k4mobidedrm.config", line 1, in <module>
ImportError: No module named PyQt4.Qt
Added 0 books in 15.1 seconds
DeDRM v6.5.3: Trying to decrypt B00FUZPQ2W_EBOK.azw
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Coldbrook
Found 4 keys to try after 0.8 seconds
Crypto Type is: 2
File is encoded with PID SxjHsLQuFZ.
Decrypting. Please wait . . . . . done
Decryption succeeded after 1.0 seconds
DeDRM v6.5.3: Finished after 1.0 seconds
Using Library AlfCrypto DLL/DYLIB/SO
Running file type plugin Kindle and Mobipocket DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.k4mobidedrm.__init__", line 68, in run
  File "calibre_plugins.k4mobidedrm.__init__", line 239, in config
  File "site-packages\calibre\customize\zipplugin.py", line 183, in load_module
  File "calibre_plugins.k4mobidedrm.config", line 1, in <module>
ImportError: No module named PyQt4.Qt
Added 0 books in 10.6 seconds
DeDRM v6.5.3: Trying to decrypt B008J2GAKU_EBOK.azw
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Batman Vol. 1: The Court of Owls (The New 52) (Batman Graphic Novel)
Found 4 keys to try after 2.8 seconds
Crypto Type is: 2
File is encoded with PID mSFc4+XCHI.
Decrypting. Please wait . . . . . done
Decryption succeeded after 4.7 seconds
DeDRM v6.5.3: Finished after 5.6 seconds
Using Library AlfCrypto DLL/DYLIB/SO
Running file type plugin Kindle and Mobipocket DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.k4mobidedrm.__init__", line 68, in run
  File "calibre_plugins.k4mobidedrm.__init__", line 239, in config
  File "site-packages\calibre\customize\zipplugin.py", line 183, in load_module
  File "calibre_plugins.k4mobidedrm.config", line 1, in <module>
ImportError: No module named PyQt4.Qt
Added Batman Vol. 1: The Court of Owls (The New 52) (Batman Graphic Novel) to db in: 3.5
Added 1 books in 15.7 seconds

**Comment by ElleKayEm** [comment]
You can remove the old "Kindle and Mobipocket DeDRM" plugin.  It does look like the DRM removal is successful.  You can't open them with the calibre ebook viewer?  If it's just a problem with conversion, try the KindleUnpack plugin.

**Comment by HorrorNerd** [comment]
no it won't open up with the viewer either

**Comment by ElleKayEm** [comment]
Have you tried importing the files again after removing the old plugin?

**Comment by mamamusings** [comment]
I'm also unable to convert an .AZW Kindle book. Made sure I was running (only) 6.5.3 of the DRM tool, restarted Calibre, deleted and reimported the book, still can't view it in the Calibre reader or convert it--both give a "this book is protected with DRM" error. Here's the log file:

calibre Debug log
calibre 2.80  embedded-python: True is64bit: True
Darwin-16.4.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '16.4.0', 'Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; root:xnu-3789.41.3~3/RELEASE_X86_64')
Python 2.7.11
OSX: ('10.12.3', ('', '', ''), 'x86_64')
Successfully initialized third party plugins: DeDRM (6, 5, 3)
devicePixelRatio: 2.0
logicalDpi: 72.0 x 72.0
physicalDpi: 128.000001922 x 127.500001915
Starting up...
Started up in 1.40 seconds with 86 books
DeDRM v6.5.3: Trying to decrypt B01FWERIOY_EBOK.azw
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Games, Design and Play: A detailed approach to iterative game design
Found 6 keys to try after 0.1 seconds
Crypto Type is: 2
DeDRM v6.5.3: Failed to decrypt with error: No key found in 6 keys tried.
DeDRM v6.5.3: Looking for new default Kindle Key after 0.1 seconds
Found k4Mac kinf2011 file: /Users/ell/Library/Containers/com.amazon.Kindle/Data/Library/Application Support/Kindle/storage/.kinf2011
Found k4Mac kindle-info file: /Users/ell/Library/Application Support/Amazon/Kindle/storage/.kindle-info
Decrypted key file using IDString 'S1K6NYADB36289' and UserName 'ell'
Couldn't decrypt file.
DeDRM v6.5.3: Found 1 new key
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Games, Design and Play: A detailed approach to iterative game design
Found 4 keys to try after 1.3 seconds
Crypto Type is: 2
DeDRM v6.5.3: Ultimately failed to decrypt after 1.3 seconds.
Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages/calibre/customize/ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 610, in run
  File "calibre_plugins.dedrm.__init__", line 560, in KindleMobiDecrypt
DeDRMError: DeDRM v6.5.3: Ultimately failed to decrypt after 1.3 seconds
Added Games, Design and Play: A detailed approach to iterative game design to db in: 0.1
Added 1 books in 1.8 seconds

**Comment by mamamusings** [comment]
I'm also running into a problem with a recently-downloaded Kindle book. Get a DRM error message when I try to view or convert the book. I'm running version 1.12 of Kindle for the Mac, no KFX files in sight. The file I imported was an .AZW format. Running (only) version 6.5.3 of the DeDRM tool. Here's the log file: 

calibre Debug log
calibre 2.80  embedded-python: True is64bit: True
Darwin-16.4.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '16.4.0', 'Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; root:xnu-3789.41.3~3/RELEASE_X86_64')
Python 2.7.11
OSX: ('10.12.3', ('', '', ''), 'x86_64')
Successfully initialized third party plugins: DeDRM (6, 5, 3)
devicePixelRatio: 2.0
logicalDpi: 72.0 x 72.0
physicalDpi: 128.000001922 x 127.500001915
Starting up...
Started up in 1.40 seconds with 86 books
DeDRM v6.5.3: Trying to decrypt B01FWERIOY_EBOK.azw
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Games, Design and Play: A detailed approach to iterative game design
Found 6 keys to try after 0.1 seconds
Crypto Type is: 2
DeDRM v6.5.3: Failed to decrypt with error: No key found in 6 keys tried.
DeDRM v6.5.3: Looking for new default Kindle Key after 0.1 seconds
Found k4Mac kinf2011 file: /Users/ell/Library/Containers/com.amazon.Kindle/Data/Library/Application Support/Kindle/storage/.kinf2011
Found k4Mac kindle-info file: /Users/ell/Library/Application Support/Amazon/Kindle/storage/.kindle-info
Decrypted key file using IDString 'S1K6NYADB36289' and UserName 'ell'
Couldn't decrypt file.
DeDRM v6.5.3: Found 1 new key
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Games, Design and Play: A detailed approach to iterative game design
Found 4 keys to try after 1.3 seconds
Crypto Type is: 2
DeDRM v6.5.3: Ultimately failed to decrypt after 1.3 seconds.
Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages/calibre/customize/ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 610, in run
  File "calibre_plugins.dedrm.__init__", line 560, in KindleMobiDecrypt
DeDRMError: DeDRM v6.5.3: Ultimately failed to decrypt after 1.3 seconds
Added Games, Design and Play: A detailed approach to iterative game design to db in: 0.1
Added 1 books in 1.8 seconds
